### PR TITLE
Support condition expressions using metric components or dataset prof…

### PR DESCRIPTION
**Pull request by richard-rogers**
…iles

## Description

This repository needs 

The commits in this pull request will 

## Changes

- Imperative commit or change title (commit ID) 
  - Add some bullet points to explain the change
- Next commit or change (commit ID)

## Related

Relates to organization/repo#number

Closes organization/repo#number

- [ ] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: python/whylogs/core/metrics/condition_count_metric.py
Comment (FelipeAdachi) [inline python/whylogs/core/metrics/condition_count_metric.py:18]: If Conditions are shared between metrics and validators, shouldn't we move it outside from `condition_count_metric.py`?

File: python/whylogs/core/metrics/condition_count_metric.py
Comment (FelipeAdachi) [inline python/whylogs/core/metrics/condition_count_metric.py]: Can you give a simple snippet on how to use this along with metricGetter or profileGetter?

File: python/whylogs/core/metrics/condition_count_metric.py
Comment (FelipeAdachi) [reply python/whylogs/core/metrics/condition_count_metric.py:18]: I see you answered this on https://github.com/whylabs/whylogs/pull/761#discussion_r959016965

We can also do this in a followup

File: python/whylogs/core/metrics/condition_count_metric.py
Comment (TheMellyBee) [inline python/whylogs/core/metrics/condition_count_metric.py]: Oh wow. THat is quite the datatype. I think it's fine for now, but this looks like a code smell that at the point this many layers... we should probably think of a data structure to make this clearer and easier to read/maintain.

File: python/whylogs/core/metrics/condition_count_metric.py
Comment (TheMellyBee) [inline python/whylogs/core/metrics/condition_count_metric.py]: Please refactor the name x to make it more clear what is being added. I see this was before you, but just as a maintenance as we are here.

**Review by TheMellyBee — DISMISSED** [review]
Looks good!

**Review by jamie256 — APPROVED** [review]
Note: I didn't explicitly dismiss Melly's approval, I added a rule to our repo that would require re-approval if you push more comits after someone approves.

I'll approve this here based on the previous approvals and the changes since then being ok.

Looks like there are some merge conflicts but I have no objections to these changes if you get a clear CI. Authors still need to resolve the conversations after addressing the concerns and answering questions.

**Comment by richard-rogers** [comment]
Going to take another swing at this

**Pull request closed without merging**
